File: backend/app/src/main/java/org/example/app/controllers/requests/search/SearchCourse.java
package org.example.app.controllers.requests.search;

import jakarta.validation.constraints.NotNull;
import lombok.Getter;
import lombok.NoArgsConstructor;

import java.time.LocalDateTime;
import java.util.UUID;

@Getter
@NoArgsConstructor
@SuppressWarnings("unused")
public class SearchCourse {
    @NotNull
    private UUID startCity;
    @NotNull
    private UUID endCity;
    @NotNull
    private LocalDateTime departureTime;
}

File: frontend/src/app/models/request-models/searchCourse.ts
export class SearchCourse {
    startCity: string;
    endCity: string;
    departureTime: Date;
    
    constructor(startCity: string, endCity: string, departureTime: Date) {
        this.startCity = startCity;
        this.endCity = endCity;
        this.departureTime = departureTime;
    }
  }
  